lib/galaxy/datatypes/text.py

```diff
@@ -117,19 +117,28 @@ class ExpressionJson(Json):
     def set_meta(self, dataset, **kwd):
         """
         """
-        json_type = "null"
-        with open(dataset.file_name) as f:
-            obj = json.load(f)
-            if isinstance(obj, int):
-                json_type = "int"
-            elif isinstance(obj, float):
-                json_type = "float"
-            elif isinstance(obj, list):
-                json_type = "list"
-            elif isinstance(obj, dict):
-                json_type = "object"
-
-        dataset.metadata.json_type = json_type
+        if dataset.has_data():
+            json_type = "null"
+            file_path = dataset.file_name
+            file_size = os.stat(file_path).st_size
+            if file_size == 0:
+                raise Exception("Invalid empty JSON encountered")
```

Comment:
`has_data` should already check the size. But should we raise that exception in the else branch ?

Reply:
Just left over debugging, I'll remove that condition.